dask/dataframe/core.py

```diff
@@ -1455,27 +1455,79 @@ def reduction(x, chunk, aggregate, token=None):
     return Scalar(merge(x.dask, dsk, dsk2), b)
 
 
-def concat(dfs):
-    """ Concatenate dataframes along rows
+def concat(dfs, respect_partitions=False):
+    """ Concatenate DataFrames along rows.
+
+    - If divisions are known, concatenate DataFrames if all divisions can be
+      concatenated (ordered).
+    - If divisions are unknown, concatenate to the last
+    - Concatenating DataFrames mixed with known and unknown divisions are not
+      supported.
+
+    Parameters
+    ----------
+
+    dfs: list
+        List of dask.DataFrames to be concatenated
+    respect_partitions: bool, default False
+        Whether to concatenate DataFrames ignoring its order. If True, every
+        divisions are concatenated each by each.
```

Comment:
Now that I think about this more, maybe rename this to `interleave_partitions`?

Explanation could be: "Interleave partitions to maintain sorted order"

If it's easy it might be nice to have an example doctest demonstrating the difference.